From 15c7050356a610caebe1cff0810912beaec21fc3 Mon Sep 17 00:00:00 2001
From: Zhao Qiang <qiang.zhao@nxp.com>
Date: Tue, 7 Feb 2017 12:54:31 +0800
Subject: [PATCH 62/74] armv8: fsl-layerscape: Add NXP LS2088A SoC support

commit f1c8da93a5544ef6e20e83e3643fbbff9a182f1b
[adjust context]

The QorIQ LS2088A processor is built on the Layerscape
architecture combining eight ARM A72 processor cores
with advanced, high-performance datapath acceleration
and network, peripheral interfaces required for
networking, telecom, wireless infrastructure, aerospace
applications and general-purpose embedded applications.

Features summary:
 - Eight 64-bit ARM v8 Cortex-A72 CPUs
 - Two 64-bit DDR4 SDRAM memory controller with ECC
 - One 32-bit DDR3 SDRAM memory controller with ECC
 - Data path acceleration architecture 2.0 (DPAA2)
 - Ethernet interfaces
 - IFC, 4 PCIe, 2 SATA, 2 USB, 1 SDXC, 2 DUARTs etc

LS2088A is compliant to the Layerscape Chassis Generation 3.

LS2088A SoC is similar to LS2085A Soc with some differences like
1)Timer controller offset is different
2)Errata A009635 and A008751 are fixed
3)Maia core

These differences can be detected via runtime during boot.
So LS2085A, LS2088A will have common u-boot image

Signed-off-by: Priyanka Jain <priyanka.jain@nxp.com>
Integrated-by: Zhao Qiang <qiang.zhao@nxp.com>
---
 arch/arm/include/asm/arch-fsl-layerscape/config.h |  3 ++
 arch/arm/include/asm/arch-fsl-layerscape/cpu.h    |  4 ++
 arch/arm/include/asm/arch-fsl-layerscape/soc.h    |  4 ++
 board/freescale/ls2080aqds/README                 | 46 ++++++++++++++++++++++-
 board/freescale/ls2080ardb/README                 | 45 +++++++++++++++++++++-
 5 files changed, 100 insertions(+), 2 deletions(-)

diff --git a/arch/arm/include/asm/arch-fsl-layerscape/config.h b/arch/arm/include/asm/arch-fsl-layerscape/config.h
index a407a30..1beec3e 100644
--- a/arch/arm/include/asm/arch-fsl-layerscape/config.h
+++ b/arch/arm/include/asm/arch-fsl-layerscape/config.h
@@ -40,6 +40,9 @@
 
 #define CONFIG_SYS_FSL_OCRAM_BASE	0x18000000	/* initial RAM */
 #define CONFIG_SYS_FSL_OCRAM_SIZE	0x00200000	/* 2M */
+#define CONFIG_FSL_TZASC_400
+#define CONFIG_FSL_TZASC_1
+#define CONFIG_FSL_TZASC_2
 
 /* DDR */
 #define CONFIG_SYS_FSL_DDR_LE
diff --git a/arch/arm/include/asm/arch-fsl-layerscape/cpu.h b/arch/arm/include/asm/arch-fsl-layerscape/cpu.h
index 10ccf54..dbbea56 100644
--- a/arch/arm/include/asm/arch-fsl-layerscape/cpu.h
+++ b/arch/arm/include/asm/arch-fsl-layerscape/cpu.h
@@ -8,6 +8,10 @@
 #define _FSL_LAYERSCAPE_CPU_H
 
 static struct cpu_type cpu_type_list[] = {
+	CPU_TYPE_ENTRY(LS2088, LS2088, 8),
+	CPU_TYPE_ENTRY(LS2084, LS2084, 8),
+	CPU_TYPE_ENTRY(LS2048, LS2048, 4),
+	CPU_TYPE_ENTRY(LS2044, LS2044, 4),
 	CPU_TYPE_ENTRY(LS2080, LS2080, 8),
 	CPU_TYPE_ENTRY(LS2085, LS2085, 8),
 	CPU_TYPE_ENTRY(LS2045, LS2045, 4),
diff --git a/arch/arm/include/asm/arch-fsl-layerscape/soc.h b/arch/arm/include/asm/arch-fsl-layerscape/soc.h
index 8691906..6cf0f5b 100644
--- a/arch/arm/include/asm/arch-fsl-layerscape/soc.h
+++ b/arch/arm/include/asm/arch-fsl-layerscape/soc.h
@@ -45,6 +45,10 @@ struct cpu_type {
 #define SVR_LS2045		0x870120
 #define SVR_LS2080		0x870110
 #define SVR_LS2085		0x870100
+#define SVR_LS2088		0x870901
+#define SVR_LS2084		0x870911
+#define SVR_LS2048		0x870921
+#define SVR_LS2044		0x870931
 
 #define SVR_MAJ(svr)		(((svr) >> 4) & 0xf)
 #define SVR_MIN(svr)		(((svr) >> 0) & 0xf)
diff --git a/board/freescale/ls2080aqds/README b/board/freescale/ls2080aqds/README
index a4abb7e..ca48cb0 100644
--- a/board/freescale/ls2080aqds/README
+++ b/board/freescale/ls2080aqds/README
@@ -2,7 +2,8 @@ Overview
 --------
 The LS2080A Development System (QDS) is a high-performance computing,
 evaluation, and development platform that supports the QorIQ LS2080A
-Layerscape Architecture processor. The LS2080AQDS provides validation and
+and LS2088A (and its personalities) Layerscape Architecture processor.
+The LS2080AQDS provides validation and
 SW development platform for the Freescale LS2080A processor series, with
 a complete debugging environment.
 
@@ -50,6 +51,49 @@ The LS2080A SoC includes the following function and features:
  - Service processor (SP) provides pre-boot initialization and secure-boot
   capabilities
 
+LS2088A SoC Overview
+------------------
+The QorIQ LS2088A processor is built on the Layerscape
+architecture combining eight ARM A72 processor cores
+with advanced, high-performance datapath acceleration
+and network, peripheral interfaces required for
+networking, telecom, wireless infrastructure, aerospace
+applications and general-purpose embedded applications.
+
+Features summary:
+ - Eight 64-bit ARM v8 Cortex-A72 CPUs
+ - Two 64-bit DDR4 SDRAM memory controller with ECC
+ - One 32-bit DDR3 SDRAM memory controller with ECC
+ - Data path acceleration architecture 2.0 (DPAA2)
+ - Ethernet interfaces
+ - IFC, 4 PCIe, 2 SATA, 2 USB, 1 SDXC, 2 DUARTs etc
+ - AIOP
+ - 8 * 1/10G + 8 *1G WRIOP
+ - L2 switch
+
+LS2088A is compliant to the Layerscape Chassis Generation 3.
+
+LS2088A SoC is similar to LS2085A Soc with some differences like
+1)Timer controller offset is different
+2)Errata A009635 and A008751 are fixed
+3)Maia core
+
+These differences can be detected via runtime during boot.
+So LS2085A, LS2088A will have common u-boot image
+
+LS2088A SoC has 3 more similar SoC personalities
+1)LS2048A, few difference w.r.t LS2088A:
+	a) Four 64-bit ARM v8 Cortex-A72 CPUs
+
+2)LS2084A, few difference w.r.t LS2088A:
+	a) No AIOP
+	b) No 32-bit DDR3 SDRAM memory
+	c) 5 * 1/10G + 5 *1G WRIOP
+	d) No L2 switch
+
+3)LS2044A, few difference w.r.t LS2084A:
+	a) Four 64-bit ARM v8 Cortex-A72 CPUs
+
  LS2080AQDS board Overview
  -----------------------
  - SERDES Connections, 16 lanes supporting:
diff --git a/board/freescale/ls2080ardb/README b/board/freescale/ls2080ardb/README
index 7fc2569..5af2957 100644
--- a/board/freescale/ls2080ardb/README
+++ b/board/freescale/ls2080ardb/README
@@ -2,7 +2,7 @@ Overview
 --------
 The LS2080A Reference Design (RDB) is a high-performance computing,
 evaluation, and development platform that supports the QorIQ LS2080A
-Layerscape Architecture processor.
+and LS2088A (and its personalitites) Layerscape Architecture processor.
 
 LS2080A SoC Overview
 ------------------
@@ -48,6 +48,49 @@ The LS2080A SoC includes the following function and features:
  - Service processor (SP) provides pre-boot initialization and secure-boot
   capabilities
 
+LS2088A SoC Overview
+------------------
+The QorIQ LS2088A processor is built on the Layerscape
+architecture combining eight ARM A72 processor cores
+with advanced, high-performance datapath acceleration
+and network, peripheral interfaces required for
+networking, telecom, wireless infrastructure, aerospace
+applications and general-purpose embedded applications.
+
+Features summary:
+ - Eight 64-bit ARM v8 Cortex-A72 CPUs
+ - Two 64-bit DDR4 SDRAM memory controller with ECC
+ - One 32-bit DDR3 SDRAM memory controller with ECC
+ - Data path acceleration architecture 2.0 (DPAA2)
+ - Ethernet interfaces
+ - IFC, 4 PCIe, 2 SATA, 2 USB, 1 SDXC, 2 DUARTs etc
+ - AIOP
+ - 8 * 1/10G + 8 *1G WRIOP
+ - L2 switch
+
+LS2088A is compliant to the Layerscape Chassis Generation 3.
+
+LS2088A SoC is similar to LS2085A Soc with some differences like
+1)Timer controller offset is different
+2)Errata A009635 and A008751 are fixed
+3)Maia core
+
+These differences can be detected via runtime during boot.
+So LS2085A, LS2088A will have common u-boot image
+
+LS2088A SoC has 3 more similar SoC personalities
+1)LS2048A, few difference w.r.t LS2088A:
+	a) Four 64-bit ARM v8 Cortex-A72 CPUs
+
+2)LS2084A, few difference w.r.t LS2088A:
+	a) No AIOP
+	b) No 32-bit DDR3 SDRAM memory
+	c) 5 * 1/10G + 5 *1G WRIOP
+	d) No L2 switch
+
+3)LS2044A, few difference w.r.t LS2084A:
+	a) Four 64-bit ARM v8 Cortex-A72 CPUs
+
  LS2080ARDB board Overview
  -----------------------
  - SERDES Connections, 16 lanes supporting:
-- 
2.1.0.27.g96db324

